File: parallel_execution/sources/single.1.f
! @@name:	single.1
! @@type:	F-fixed
! @@operation:	link
! @@expect:	success
! @@version:	pre_omp_3.0
      SUBROUTINE WORK1()
      END SUBROUTINE WORK1

      SUBROUTINE WORK2()
      END SUBROUTINE WORK2

      PROGRAM SINGLE_EXAMPLE
!$OMP PARALLEL

!$OMP SINGLE
        print *, "Beginning work1."
!$OMP END SINGLE

        CALL WORK1()

!$OMP SINGLE
        print *, "Finishing work1."
!$OMP END SINGLE

!$OMP SINGLE
        print *, "Finished work1 and beginning work2."
!$OMP END SINGLE NOWAIT

        CALL WORK2()

!$OMP END PARALLEL

      END PROGRAM SINGLE_EXAMPLE

File: parallel_execution/sources/single.1.c
/*
* @@name:	single.1
* @@type:	C
* @@operation:	link
* @@expect:	success
* @@version:	pre_omp_3.0
*/
#include <stdio.h>

void work1() {}
void work2() {}

int main()
{
  #pragma omp parallel
  {
    #pragma omp single
      printf("Beginning work1.\n");

    work1();

    #pragma omp single
      printf("Finishing work1.\n");

    #pragma omp single nowait
      printf("Finished work1 and beginning work2.\n");

    work2();
  }
}
